nilearn/datasets/atlas.py

```diff
@@ -576,3 +577,82 @@ def fetch_yeo_2011_atlas(data_dir=None, url=None, resume=True, verbose=1):
         url=url,
         resume=resume,
         verbose=verbose)
+
+
+def fetch_atlas_aal_spm_12(data_dir=None, url=None, resume=True, verbose=1):
+    """Downloads and returns the AAL template for SPM 12.
+
+    The provided images are based on the spatially normalized single-subject
+    high-resolution T1 volume provided by the Montreal Neurological Institute
+    (MNI)
+
+    Parameters
+    ----------
+    data_dir: string
+        directory where data should be downloaded and unpacked.
+
+    url: string
+        url of file to download.
+
+    resume: bool
+        whether to resumed download of a partly-downloaded file.
+
+    verbose: int
+        verbosity level (0 means no message).
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        dictionary-like object, keys are:
+
+        - "regions": str. path to nifti file containing regions.
+
+        - "labels": dict. labels dictionary with their region id as key and
+                    name as value
+
+    Notes
+    -----
+    For more information on this dataset's structure, see
+    http://www.gin.cnrs.fr/AAL-217?lang=en
+
+    Automated Anatomical Labeling of Activations in SPM Using a Macroscopic
+    Anatomical Parcellation of the MNI MRI Single-Subject Brain.
+    N. Tzourio-Mazoyer, B. Landeau, D. Papathanassiou, F. Crivello,
+    O. Etard, N. Delcroix, B. Mazoyer, and M. Joliot.
+    NeuroImage 2002. 15 :273-28
+
+    Licence: unknown.
+    """
+    spm_version = 12
+    if url is None:
+        baseurl = "http://www.gin.cnrs.fr/AAL/aal_for_SPM%i.tar.gz"
+        url = baseurl % spm_version
+    opts = {'uncompress': True}
+
+    dataset_name = "aal_spm_12"
+    # keys and basenames would need to be handled for each spm_version
+    # for now spm_version 12 is hardcoded.
+    keys = ("regions", "labels")
+    basenames = (
+        "AAL.nii", "AAL.xml")
+
+    filenames = [(os.path.join("aal_for_SPM%i" % spm_version, f), url, opts)
+                 for f in basenames]
+
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+    atlas_img, labels_file = _fetch_files(data_dir, filenames, resume=resume,
+                                          verbose=verbose)
+
+    fdescr = _get_dataset_descr(dataset_name)
```

Comment:
Would it make sense to add a description of the dataset in nilearn/datasets/description? All you need to do is add aal_spm_12.rst there I think.